include/NeoFOAM/core/primitives/vector.hpp

```diff
@@ -145,9 +145,9 @@ Vector operator*(const scalar& sclr, Vector rhs)
 }
 
 KOKKOS_INLINE_FUNCTION
-Vector operator&(const Vector& lhs, Vector rhs)
+scalar operator&(const Vector& lhs, Vector rhs)
```

Comment:
Maybe it makes sense to have both? preferably i would say this should be implemented in a free function.
```suggestion
/* @brief dot product */
scalar operator*(const Vector& lhs, Vector rhs) {return dotProduct(lhs, rhs)}

/* @brief inner product */
Vector operator&(const Vector& lhs, Vector rhs) {return innerProduct(lhs, rhs)}
```

Reply:
This would be inconsistent with the Foam API.  Also the dotProduct reduces two vector to a scalar. An alternative name for the dotProduct is innerProduct